includes/services/extended/linkedin.php

```diff
@@ -1,97 +1,153 @@
+		return $scope;
 	}
 
-	function member_permissions( $permissions = '' ) {
-		$permissions = 'r_basicprofile';
-		return $permissions;
+	/**
+	 * By adding the `x-li-format: json` header here, we can avoid having to append `?format=json` to all requests.
+	 *
+	 * https://developer.linkedin.com/docs/rest-api#hero-par_longformtext_4_longform-text-content-par_resourceparagraph
+	 *
+	 * @param string $url
+	 * @param array $params
+	 * @return array|Keyring_Error|mixed|object|string
+	 */
+	function request( $url, array $params = array() ) {
+		$params['headers']['x-li-format'] = 'json';
+		return parent::request( $url, $params );
 	}
 
+	/**
+	 * Build the meta for the token.
+	 *
+	 * @param mixed $token
+	 * @return false|int|mixed
+	 */
 	function build_token_meta( $token ) {
-		// Set the token so that we can make requests using it
 		$this->set_token(
 			new Keyring_Access_Token(
 				$this->get_name(),
-				new OAuthToken(
-					$token['oauth_token'],
-					$token['oauth_token_secret']
-				)
+				$token['access_token'],
+				array()
 			)
 		);
 
-		// Get user profile information
-		$response = $this->request( "https://api.linkedin.com/v1/people/~:(id,formatted-name,picture-url)?format=json" );
+		$response = $this->request(
+			$this->self_url . '?projection=(id,firstName,lastName,profilePicture(displayImage~:playableStreams))',
+			array( 'method' => $this->self_method )
+		);
+
 		if ( Keyring_Util::is_error( $response ) ) {
 			$meta = array();
 		} else {
 			$this->person = $response;
+
+			$firstName = $this->person->firstName;
+			$lastName  = $this->person->lastName;
+			$lfirst = "{$firstName->preferredLocale->language}_{$firstName->preferredLocale->country}";
+			$llast  = "{$lastName->preferredLocale->language}_{$lastName->preferredLocale->country}";
+
+			$profilePicture = $this->person->profilePicture;
+
 			$meta = array(
 				'user_id' => $this->person->id,
-				'name'    => $this->person->formattedName,
-				'picture' => $this->person->pictureUrl,
+				'name'    => $firstName->localized->{$lfirst} . ' ' . $lastName->localized->{$llast},
+				'picture' => $profilePicture->{'displayImage~'}->elements[0]->identifiers[0]->identifier,
 			);
 		}
 
-		return apply_filters( 'keyring_access_token_meta', $meta, $this->get_name(), $token, $response, $this );
+		return apply_filters( 'keyring_access_token_meta', $meta, self::NAME, $token, $response, $this );
 	}
 
 	function get_display( Keyring_Access_Token $token ) {
 		return $token->get_meta( 'name' );
 	}
 
-	function test_connection() {
-			$res = $this->request( "https://api.linkedin.com/v1/people/~:(id,formatted-name)?format=json" );
-			if ( ! Keyring_Util::is_error( $res ) ) {
-				return true;
-			}
+	/**
+	 * Get profile picture.
+	 *
+	 * @return string|mixed
+	 */
+	function fetch_profile_picture () {
+		$response = $this->request(
+			$this->self_url . '?projection=(profilePicture(displayImage~:playableStreams))',
+			[ 'method' => $this->self_method ]
```

Comment:
I believe this is the first usage of short-array syntax in Keyring. Should be fine since that syntax has been around forever at this point, just noting.